tests/test_connection.py

```diff
@@ -129,8 +136,10 @@ async def wrapper(*args, **kwargs):
 @attr.s(hash=False, eq=False)
 class MemoryListener(trio.abc.Listener):
     closed = attr.ib(default=False)
-    accepted_streams = attr.ib(factory=list)
-    queued_streams = attr.ib(factory=lambda: trio.open_memory_channel(1))
+    accepted_streams: list[
+        tuple[trio.abc.SendChannel[str], trio.abc.ReceiveChannel[str]]
+    ] = attr.ib(factory=list)
```

Comment:
Python < 3.9 doesn't support direct use of `list`, etc. for type annotations

Reply:
It works because I added `from __future__ import annotations`, otherwise the CI test would fail.